bootenv/env.go

```diff
@@ -0,0 +1,39 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2023 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package bootenv
+
+import (
+	"sync/atomic"
+
+	"github.com/snapcore/secboot"
+)
+
+var (
+	currentModel    atomic.Value
+	currentBootMode atomic.Value
+)
+
+var SetModel = func(model secboot.SnapModel) bool {
```

Comment:
The initial branch only allowed `currentModel` and `currentBootMode` to be updated once (using `CompareAndSwap(nil, model)`) so I couldn't change their values between tests. Does that need to change or did I miss something here?

Reply:
I still can't see how I can workaround the fact that atomic.Value can't be set to nil after it is set. My intention here was to mock both the setters and getters to use `keyDataPlatformSuite` variables instead of the real ones. 

I tried changing the setters to use `atomic.Value.Store` then have a new function `ClearModelAndBootMode` in `export_test.go` but `atomic.value.Store`, `atomic.value.Swap` and `atomic.value.CompareAndSwap` operations fail when trying to set a nil value to an atomic variable that has been set.

Reply:
Ah, when I suggested just setting `currentModel` and `currentBootMode` to zero values, I meant not doing it atomically and just doing something like this:
```
currentModel = atomic.Value{}
currentBootMode = atomic.Value{}
```

This obviously isn't safe in production code, but should be fine from test code.

Reply:
Ah right, thanks for this suggestion, it cleaned up the code quite a bit. I implemented it as suggested.